VocaDb.Migrations/Migrations.cs

```diff
@@ -1,10 +1,157 @@
+
+			// Rename columns on `AlbumReviews` table.
+			Rename.Column("Date").OnTable(TableNames.AlbumReviews).To("Created");
+			Rename.Column("Text").OnTable(TableNames.AlbumReviews).To("Message");
+			Rename.Column("User").OnTable(TableNames.AlbumReviews).To("Author");
+			Delete.Index("UX_AlbumReviews").OnTable(TableNames.AlbumReviews);
+			Create.Index().OnTable(TableNames.AlbumReviews)
+				.OnColumn("Album").Ascending()
+				.OnColumn("Author").Ascending()
+				.OnColumn("LanguageCode").Ascending();
+
+			// Create `Comments` table.
+			Create.Table(TableNames.Comments)
+				.WithColumn("Id").AsInt32().NotNullable().Identity().PrimaryKey()
+				.WithColumn("Author").AsInt32().NotNullable().ForeignKey(TableNames.Users, "Id").OnDelete(Rule.Cascade)
+				.WithColumn("Created").AsDateTime().NotNullable()
+				.WithColumn("Deleted").AsBoolean().NotNullable().WithDefaultValue(false)
+				.WithColumn("Message").AsString(int.MaxValue).NotNullable()
+				// TODO: remove these columns
+				.WithColumn("OldTable").AsString().Nullable()
+				.WithColumn("OldId").AsInt32().Nullable();
+
+			var commentTables = new[]
+			{
+				new CommentTable(name: TableNames.AlbumComments, entryTypeName: "Album"),
+				new CommentTable(name: TableNames.ArtistComments, entryTypeName: "Artist"),
+				new CommentTable(name: TableNames.DiscussionComments, schema: SchemaNames.Discussions, entryTypeName: "Topic"),
+				new CommentTable(name: TableNames.ReleaseEventComments, entryTypeName: "ReleaseEvent"),
+				new CommentTable(name: TableNames.SongComments, entryTypeName: "Song"),
+				new CommentTable(name: TableNames.SongListComments, entryTypeName: "SongList"),
+				new CommentTable(name: TableNames.TagComments, entryTypeName: "Tag"),
+				new CommentTable(name: TableNames.UserComments, entryTypeName: "User"),
+
+				new CommentTable(name: TableNames.DiscussionTopics, schema: SchemaNames.Discussions),
+				new CommentTable(name: TableNames.AlbumReviews, entryTypeName: "Album"),
+			};
+
+			// insert into Comments(OldTable, OldId, Author, Created, Message)
+			//     select 'AlbumComments', Id, Author, Created, Message from AlbumComments union
+			//     select 'ArtistComments', Id, Author, Created, Message from ArtistComments union
+			//     select 'discussions.DiscussionComments', Id, Author, Created, Message from discussions.DiscussionComments union
+			//     ...
+			//     select 'UserComments', Id, Author, Created, Message from UserComments union
+			//     select 'DiscussionTopics', Id, Author, Created, Message from discussions.DiscussionTopics union
+			//     select 'AlbumReviews', Id, Author, Created, Message from AlbumReviews
+			// order by Created
+			var values = string.Join(" union ", commentTables.Select(table => $"select '{table.NameWithSchema}', Id, Author, Created, Message from {table.NameWithSchema}"));
+			Execute.Sql($@"insert into {TableNames.Comments}(OldTable, OldId, Author, Created, Message) {values} order by Created");

```

Comment:
I am perplexed and concerned that there is a need to run SQL. Is this how this normally works, or is there a special need here?

Reply:
For migrations, unless it's a basic update or insert, you may need to write SQL. As far as I know, there is no way to do update like this using FluentMigrator without SQL. But it's ok, sometimes SQL is the easiest.